org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BundleUtils.java

```diff
@@ -153,12 +178,34 @@ public static void loadBundles(Collection<String> bundleLocations) throws CoreEx
 				status.add(new Status(IStatus.ERROR, context.getBundle().getSymbolicName(), "Cannot extract bundle symbolicName or version " + bundleLocation, e));
 			}
 		}
+
+		refreshBundles(toRefresh, frameworkWiring);
+
 		status.addAll(startBundles(bundlesToStart));
 		if (status.getChildren().length > 0) {
 			throw new CoreException(status);
 		}
 	}
 
+	private static void refreshBundles(Set<Bundle> toRefresh, FrameworkWiring frameworkWiring) {
+		if (!toRefresh.isEmpty()) {
+			JavaLanguageServerPlugin.logInfo("Refresh the bundles");
+			final CountDownLatch latch = new CountDownLatch(1);
+			frameworkWiring.refreshBundles(toRefresh, new FrameworkListener() {
+				@Override
+				public void frameworkEvent(FrameworkEvent event) {
+					latch.countDown();

```

Comment:
Need check the specific finish event such as PACKAGES_REFRESHED, ERROR, STOPPED ?

Reply:
Yes, I should add `event.getType() == FrameworkEvent.PACKAGES_REFRESHED` check